### PR TITLE
BigQuery: Increment version to 0.28.0

### PR DESCRIPTION
**Pull request by tswast**
Prepare for beta 2 launch.

Note: this release depends on a google-cloud-core 0.28.0 release happening.

### CODE REVIEW

**Comment by dhermes** [comment]
@tswast Does this release depend on un-released features of `google-cloud-core`?

**Comment by tswast** [comment]
Well, it does use the `api_core` package now. I figured that'd be a breaking change, but maybe it doesn't have to be synced with that release?

**Comment by dhermes** [comment]
@tswast Let's just make sure we're all good before pushing to the real PyPI? I can assist you in pushing to TestPyPI if you like.

**Comment by tswast** [comment]
Yeah, I'll probably need some help with this. I'll pick this up again tomorrow when Jon is planning on doing a core release.

**Comment by dhermes** [comment]
Good. We should also make sure to resolve #4186 when `google-cloud-core` gets released.

**Comment by lukesneeringer** [comment]
I want to be _really_ careful about a core release. It almost always involves a release of the world that breaks things.

**Comment by tswast** [comment]
How might I go about patching so that it doesn't use `api_core` so that we aren't synchronized to a core release? Or do we think chances are good it will go smooth today?

**Comment by dhermes** [comment]
@lukesneeringer We have to do at least one more "release the world", so we might as well rip the band-aid.

**Comment by tswast** [comment]
I added a `CHANGELOG.md` file, following Jon's lead in https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4266 as a draft of the release notes.

File: bigquery/setup.py [hunk 1]
```diff
@@ -52,14 +52,15 @@
 
 REQUIREMENTS = [
     'google-cloud-core >= 0.28.0, < 0.29dev',
+    'google-api-core >= 0.1.1, < 0.2.0dev',
```
Comment (dhermes) [inline bigquery/setup.py]: @jonparrott This is what you had in mind, adding an explicit dep on `google-api-core`?

**Comment by jonparrott** [comment]
Yes, this is fine. It's not needed right now but since bq explicitly
depends on API core it is cool to go ahead and do it. In the future we can
drop the cloud core dep.

**Pull request merged**